src/backend/utils/cache/plancache.c

```diff
@@ -542,18 +545,18 @@ RevalidateCachedQuery(CachedPlanSource *plansource, IntoClause *intoClause)
 	 */
 	if (intoClause)
 	{
-		CreateTableAsStmt *ctas_stmt;
+		SelectStmt *select;
 
```

Comment:
why do we want to use SelectStmt instead of CreateTableAsStmt?

Reply:
Currently only Select is supported as the query for CTAS.

Reply:
Hi @yydzero, what we want to re-plan at this place is the SelectStmt, rather than the CreateTableAsStmt.  Currently for CTAS, Postgres only supports the prepared statement to be a select statement. 
And from implementation view, if we use CreateTableAsStmt here, the result of the revalidation would be an UtilityStmt, and then we will encounter error in function `ExecuteQuery()`, because it is expecting the prepared statement to be a SELECT.
```
        pstmt = (PlannedStmt *) linitial(plan_list);
        if (!IsA(pstmt, PlannedStmt) ||
            pstmt->commandType != CMD_SELECT ||
            pstmt->utilityStmt != NULL)
            ereport(ERROR,
                    (errcode(ERRCODE_WRONG_OBJECT_TYPE),
                     errmsg("prepared statement is not a SELECT")));
```

Reply:
any possibility to make revalidation's result not an UtilityStmt?

or change pstmt->commandType != CMD_SELECT to 'pstmt->commandType != CMD_SELECT OR pstmt->commandType != CMD_UTILITY' for CTAS case?

does this make sense?

Reply:
This may work but will diverge from upstream.  Currently PG doesn't support prepare a utility statement. So if we retrieve the plan from plancache and re-plan it, we are expecting it to be not an utility.